libflux/flux-core/src/semantic/tests.rs

```diff
@@ -3655,16 +3692,27 @@ error @3:17-3:18: undefined identifier y",
 
 #[test]
 fn primitive_kind_errors() {
-    test_error_msg! {
+    test_pretty_error_msg! {
         env: map![
             "isType" => "(v: A, type: string) => bool where A: Basic",
         ],
         src: r#"
             isType(v: {}, type: "record")
             isType(v: [], type: "array")
         "#,
-        err: "error @2:13-2:42: {} is not Basic (argument v)
+        err: expect_test::expect![[r#"
+            error: 
+              ┌─ :2:13
+              │
+            2 │             isType(v: {}, type: "record")
+              │             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ {} is not Basic (argument v)
```

Comment:
Due to how constraints are generated for call expressions we unfortunately do not get more precise information atm, should be fixable though.

Reply:
The formatting is really nice. I wonder about how errors are rendered in the UI though, is there a way to get the more concise message?

Reply:
If it's not possible to present a multi-line error through the UI, I'd bet we could branch the reporting based on a config option.

Reply:
Yep, we can just branch on it and omit the source display. codespan has some support for alternate displays as well https://docs.rs/codespan-reporting/0.11.1/codespan_reporting/term/enum.DisplayStyle.html